stdlib/auth/src/main/ballerina/auth/ldap_auth_store_provider.bal

```diff
@@ -0,0 +1,147 @@
+import ballerina/crypto;
+import ballerina/runtime;
+
+# Represents configurations that required for LDAP auth store.
+#
+# + domainName - Unique name to identify the user store
+# + connectionURL - Connection URL to the user store server
+# + connectionName - The username used to connect to the user store
+# + connectionPassword - Password for the ConnectionName user
+# + userSearchBase - DN of the context or object under which the user entries are stored in the user store
+# + userEntryObjectClass - Object class used to construct user entries
+# + userNameAttribute - The attribute used for uniquely identifying a user entry
+# + userNameSearchFilter - Filtering criteria used to search for a particular user entry
+# + userNameListFilter - Filtering criteria for searching user entries in the user store
+# + groupSearchBase - DN of the context or object under which the group entries are stored in the user store
+# + groupEntryObjectClass - Object class used to construct group entries
+# + groupNameAttribute - The attribute used for uniquely identifying a group entry
+# + groupNameSearchFilter - Filtering criteria used to search for a particular group entry
+# + groupNameListFilter - Filtering criteria for searching group entries in the user store
+# + membershipAttribute - Define the attribute that contains the distinguished names (DN) of user objects that are in a group
+# + userRolesCacheEnabled -  To indicate whether to cache the role list of a user
+# + connectionPoolingEnabled - Define whether LDAP connection pooling is enabled
+# + ldapConnectionTimeout - Timeout in making the initial LDAP connection
+# + readTimeout - The value of this property is the read timeout in milliseconds for LDAP operations
+# + retryAttempts - Retry the authentication request if a timeout happened
+# + secureClientSocket - The SSL configurations for the ldap client socket. This needs to be configured in order to
+#                  communicate through ldaps.
+public type LdapAuthProviderConfig record {
+    string domainName;
+    string connectionURL;
+    string connectionName;
+    string connectionPassword;
+    string userSearchBase;
+    string userEntryObjectClass;
+    string userNameAttribute;
+    string userNameSearchFilter;
+    string userNameListFilter;
+    string[] groupSearchBase;
+    string groupEntryObjectClass;
+    string groupNameAttribute;
+    string groupNameSearchFilter;
+    string groupNameListFilter;
+    string membershipAttribute;
+    boolean userRolesCacheEnabled = false;
+    boolean connectionPoolingEnabled = true;
+    int ldapConnectionTimeout = 5000;
+    int readTimeout = 60000;
+    int retryAttempts = 0;
+    SecureClientSocket? secureClientSocket;
+    !...
+};
+
+# Configures the SSL/TLS options to be used for LDAP communication.
+#
+# + trustStore - Configures the trust store to be used
+# + trustedCertFile - A file containing a list of certificates or a single certificate that the server trusts
```

Comment:
Shouldn't this be the list of certificates that 'client' trusts, not the 'server'?

Reply:
Yes, It should be the client, Thanks